File: sirius-ruby-client-web/lib/sirius/web/classes/web_popup_list.rb
module Sirius
  module Client
    module Web
      module Classes
        class WebPopupList < WebControl
          def initialize(parent, locator, parentElement = nil)
            super(parent, locator, parent_element)
          end

          def multi?
            client.core.isMultiple(@parent_element, @locator)
          end

          def deselect_all
            client.core.deselectAll(@parent_element, @locator)
          end

          def deselect_by_index(index)
            client.core.deselectByIndex(@parent_element, @locator, index)
          end

          def  deselect_by_value(value)
            client.core.deselectByValue(@parent_element, @locator, value)
          end

          def deselect_by_visible_text(text)
            client.core.deselectByVisibleText(@parent_element, @locator, text)
          end

          def get_all_options
            client.core.getAllOptions(@parent_element, @locator)
          end

          def select_by_index(index)
            client.core.selectByIndex(@parent_element, @locator, index)
          end

          def select_by_value(value)
            client.core.selectByValue(@parent_element, @locator, value)
          end

          def select_by_visible_text(text)
            client.core.selectByVisibleText(@parent_element, @locator, text)
          end

          def get_all_selected_options
            client.core.getAllSelectedOptions(@parent_element, @locator)
          end

          def get_first_selected_option
            client.core.getFirstSelectedOption(@parent_element, @locator)
          end
        end
      end
    end
  end
end

File: sirius.java.client.web/src/main/java/org/sirius/client/web/classes/WebPopupList.java
/**
 * .
 */
package org.sirius.client.web.classes;

import java.rmi.RemoteException;

/**
 * @author Myk Kolisnyk .
 */
public class WebPopupList extends WebControl {

	/**
	 * @param parent
	 *            .
	 * @param locator
	 *            .
	 */
	public WebPopupList(final Frame parent, final String locator) {
		super(parent, locator);
	}

	/**
	 * @param parent
	 *            .
	 * @param parentElement
	 *            .
	 * @param locator
	 *            .
	 */
	public WebPopupList(final Frame parent, final String parentElement,
			final String locator) {
		super(parent, parentElement, locator);
	}

	/**
	 * .
	 * 
	 * @throws RemoteException .
	 */
	public final void deselectAll() throws RemoteException {
		this.client().core().deselectAll(getParentElement(), getLocator());
	}

	/**
	 * .
	 * 
	 * @param arg3
	 *            .
	 * @throws RemoteException .
	 */
	public final void deselectByIndex(final int arg3) throws RemoteException {
		this.client().core()
				.deselectByIndex(getParentElement(), getLocator(), arg3);
	}

	/**
	 * .
	 * 
	 * @param arg3
	 *            .
	 * @throws RemoteException .
	 */
	public final void deselectByValue(final String arg3) throws RemoteException {
		this.client().core()
				.deselectByValue(getParentElement(), getLocator(), arg3);
	}

	/**
	 * .
	 * 
	 * @param arg3
	 *            .
	 * @throws RemoteException .
	 */
	public final void deselectByVisibleText(final String arg3)
			throws RemoteException {
		this.client().core()
				.deselectByVisibleText(getParentElement(), getLocator(), arg3);
	}

	/**
	 * .
	 * 
	 * @return .
	 * @throws RemoteException .
	 */
	public final String[] getAllOptions() throws RemoteException {
		return this.client().core()
				.getAllOptions(getParentElement(), getLocator());
	}

	/**
	 * .
	 * 
	 * @return .
	 * @throws RemoteException .
	 */
	public final String[] getAllSelectedOptions() throws RemoteException {
		return this.client().core()
				.getAllSelectedOptions(getParentElement(), getLocator());
	}

	/**
	 * .
	 * 
	 * @return .
	 * @throws RemoteException .
	 */
	public final String getFirstSelectedOption() throws RemoteException {
		return this.client().core()
				.getFirstSelectedOption(getParentElement(), getLocator());
	}

	/**
	 * .
	 * 
	 * @return .
	 * @throws RemoteException .
	 */
	public final boolean isMulti() throws RemoteException {
		return this.client().core()
				.isMultiple(getParentElement(), getLocator());
	}

	/**
	 * .
	 * 
	 * @param arg3
	 *            .
	 * @throws RemoteException .
	 */
	public final void selectByIndex(final int arg3) throws RemoteException {
		this.client().core()
				.selectByIndex(getParentElement(), getLocator(), arg3);
	}

	/**
	 * .
	 * 
	 * @param arg3
	 *            .
	 * @throws RemoteException .
	 */
	public final void selectByValue(final String arg3) throws RemoteException {
		this.client().core()
				.selectByValue(getParentElement(), getLocator(), arg3);
	}

	/**
	 * .
	 * 
	 * @param arg3
	 *            .
	 * @throws RemoteException .
	 */
	public final void selectByVisibleText(final String arg3)
			throws RemoteException {
		this.client().core()
				.selectByVisibleText(getParentElement(), getLocator(), arg3);
	}
}
